### DISCUSSION TITLE
Is there a way to get Audio CMSampleBuffer?

### DISCUSSION THREAD

**Discussion by TalDabush** [HaishinKit/HaishinKit.swift#1039] [Q&A]
Hi, 
Im trying to get the audio sample buffers.
I tried to:

1. get AVaudioBuffer via RTMPStream delegate using this function: 
  rtmpStream(_ stream: RTMPStream, didOutput audio: AVAudioBuffer, presentationTimeStamp: CMTime)
  but it never executes.
  (after some debugging I realized that the RTMPMuxer is the delegate of the AudioCodec and not the AVMixer which eventually triggers the RTMPStream delegate)

2. I saw this [https://github.com/shogo4405/HaishinKit.swift/issues/575](url)
but unlike VideoEffect, the AudioEffect execute params doesn't include the sample
is there a way to get the sample from the UnsafeMutableAudioBufferListPointer?

is there another way to get the audio sample?

Thanks in advance,

**Comment by shogo4405** [comment]
Hi. Yes you can get the UnsafeMutableAudioBufferListPointer.
- `stream.registerAudioEffect(_ effect: AudioEffect)`
- The AudioEffect interfaces are https://github.com/shogo4405/HaishinKit.swift/blob/9406e5de007e833b1bf40259051767522a74ab6f/Sources/Media/AudioEffect.swift

**Comment by TalDabush** [comment]
Hi, thank you for replying.
I saw that I can use 
`stream.registerAudioEffect(_ effect: AudioEffect)`
to get UnsafeMutableAudioBufferListPointer,
what Im asking is -
Is there a way to get the `CMSampleBuffer` from the `UnsafeMutableAudioBufferListPointer` ?

**Comment by shogo4405** [comment]
OK. No way.
